starfish/spots/_detector/local_search_blob_detector.py

```diff
@@ -0,0 +1,423 @@
+        to match spots up across rounds. In sparse conditions, the search can be seeded in one round
+        with a relatively large radius. In crowded images, the spots can be seeded in all rounds and
+        consensus filtering can be used to extract codes that are consistently extracted across
+        rounds.
+
+        Parameters
+        ----------
+        min_sigma : float
+            The minimum standard deviation for Gaussian Kernel. Keep this low to
+            detect smaller blobs.
+        max_sigma : float
+            The maximum standard deviation for Gaussian Kernel. Keep this high to
+            detect larger blobs.
+        threshold : float
+            The absolute lower bound for scale space maxima. Local maxima smaller
+            than thresh are ignored. Reduce this to detect blobs with less
+            intensities.
+        detector_method : str ['blob_dog', 'blob_log']
+            Name of the type of detection method used from skimage.feature, default: blob_log.
+        search_radius : int
+            Number of pixels over which to search for spots in other rounds and channels.
+        anchor_round : int
+            The imaging round against which other rounds will be checked for spots in the same
+            approximate pixel location.
+        detector_kwargs : Dict[str, Any]
+            Additional keyword arguments to pass to the detector_method.
+        """
+        self.min_sigma = min_sigma
+        self.max_sigma = max_sigma
+        self.threshold = threshold
+        self.is_volume = True  # TODO test 2-d spot calling
+        self.exclude_border = exclude_border
+        self.search_radius = search_radius
+        self.anchor_round = anchor_round
+        self.detector_kwargs = detector_kwargs
+        try:
+            self.detector_method = blob_detectors[detector_method]
+        except ValueError:
+            raise ValueError(f"Detector method must be one of {list(blob_detectors.keys())}")
+
+    def _spot_finder(self, data: xr.DataArray) -> pd.DataFrame:
+        """Find spots in a data volume.
+
+        Parameters
+        ----------
+        data : xr.DataArray
+            3D (z, y, x) data volume in which spots will be identified.
+
+        Returns
+        -------
+        pd.DataFrame
+            DataFrame wrapping the output of skimage blob_log or blob_dog with named outputs.
+
+        """
+
+        # results = np.ndarray: n_spots x (z, y, x, radius, sigma_z, sigma_y, sigma_x)
+        results = self.detector_method(
+            data,
+            min_sigma=self.min_sigma,
+            max_sigma=self.max_sigma,
+            threshold=self.threshold,
+            exclude_border=self.exclude_border,
+            **self.detector_kwargs
+        )
+
+        # if spots were detected
+        if results.shape[0]:
+
+            # measure intensities
+            z_inds = results[:, 0].astype(int)
```

Comment:
Does this not work if results is empty?  I didn't look over this code super carefully but it doesn't seem like anything here strictly requires results be non-empty.